gr-digital/lib/constellation.cc

```diff
@@ -266,13 +278,17 @@ std::vector<float> constellation::calc_soft_dec(gr_complex sample, float npwr)
     std::vector<float> tmp(2 * k, 0);
     std::vector<float> s(k, 0);
 
+    //check if noise power was overriden
+    if (npwr != d_npwr && npwr != 1.0){d_npwr = npwr;}
+    else {npwr = d_npwr;}
+
     for (int i = 0; i < M; i++) {
         // Calculate the distance between the sample and the current
         // constellation point.
-        float dist = std::abs(sample - d_constellation[i]);
+        float dist = powf(std::abs(sample - d_constellation[i]), 2.0f);
```

Comment:
```suggestion
        const float dist = std::abs(sample - d_constellation[i]);
        const float dist_sq = dist * dist;
```
Using `powf()` to compute a square seems an overkill. Also, calling `dist` to a variable which is actually the distance squared could be confusing. This fixes both problems. I've also taken the liberty to make these variables `const`.